### DISCUSSION TITLE
How to speed up forge by holding more models/controlnet in cache etc - Rec Settings for 24GB VRAM

### DISCUSSION THREAD

**Discussion by a3nima** [lllyasviel/stable-diffusion-webui-forge#575] [Q&A]
I've noticed that my A1111 is faster, when I use IP Adapter Controlnet in image generation (+lora) and hiresfix. 
While forge seems to create and load everything new by every run, even though I dont change the controlnet input/settings, A1111 seems to hold the models in cache and therefore is faster. 

So what do would you suggest to set up in my forge settings, to speed up that process ? (24GB VRAM/Windows10)

For example should I change these settings:
![grafik](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/44588648/24263c4b-8428-47a0-afdb-022450922f51)

![grafik](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/44588648/b31dde12-d18d-43d1-b0a4-fe402c701836)

![grafik](https://github.com/lllyasviel/stable-diffusion-webui-forge/assets/44588648/53c6a9bc-8b13-4090-babc-7385326c53e8)

edit: strange enough, today the times are looking better and I'm not sure about my post regarding a1111 vs forge anymore. still: before hiresfix there is a delay. maybe someone can recommend settings for maximum speed if you are not close to the edge of OOM and can afford to keep models and other things cached. Tia :)

**Comment by pflky** [comment]
It will always swap stuff from RAM to VRAM. Your issue is probably that since you're not caching anything, you're having to load it from hard drive. I have a 7GB/s NVME drive and so reading large files without caching isn't an issue. With 24GB of VRAM, I would just set it to cache everything if you're not going to get close to running out of memory. So I'd set it to cache the maximum number of controlnets you think you'll use, and cache the VAE (you only need it to cache 1), and how many every Lora's you might use at once or during a session. I'd still only keep one model on the device, as that's going to be the biggest VRAM cost.
